### ISSUE TITLE
TypeError: this.readConfig is not a function

### ISSUE THREAD

**Issue by andressrg** [ember-cli-deploy/ember-cli-deploy-redis#64]
`TypeError: this.readConfig is not a function` is getting thrown when deploying.

```
TypeError: this.readConfig is not a function
TypeError: this.readConfig is not a function
    at Object.DeployPluginBase.extend.defaultConfig.redisDeployClient (/Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli-deploy-lightning-pack/node_modules/ember-cli-deploy-redis/index.js:55:36)
    at CoreObject.extend.readConfig (/Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli-deploy-lightning-pack/node_modules/ember-cli-deploy-redis/node_modules/ember-cli-deploy-plugin/index.js:59:30)
    at CoreObject.DeployPluginBase.extend.fetchInitialRevisions (/Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli-deploy-lightning-pack/node_modules/ember-cli-deploy-redis/index.js:137:38)
    at Object._pipeline.register.fn (/Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli-deploy/lib/tasks/pipeline.js:93:21)
    at Pipeline._notifyPipelinePluginHookExecution (/Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli-deploy/lib/models/pipeline.js:175:19)
    at tryCatch (/Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:538:12)
    at invokeCallback (/Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:553:13)
    at /Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:628:16
    at flush (/Users/andres/Developer/take2mfg/take2-shop/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:2373:5)
```

Appears to be related to this change https://github.com/ember-cli-deploy/ember-cli-deploy-plugin/pull/15

**Comment by achambers** [comment]
Hi @andressrg Can you please post your `config/deploy.js` ?

**Comment by blimmer** [comment]
This happens to me too, here's my config:

```js
/* eslint-env node */
const _merge = require('lodash/object/merge');

module.exports = function(deployTarget) {
  const ENV = {
    build: {},
    redis: { allowOverwrite: true },
    s3:    {
      accessKeyId: 'redacted',
      secretAccessKey: process.env.S3_SECRET_ACCESS_KEY,
      bucket: 'redacted',
      region: 'us-east-1',
    },
    hipchat: {
      authToken:       process.env.HIPCHAT_AUTH_TOKEN,
      roomNotifyToken: process.env.HIPCHAT_ROOM_NOTIFY_TOKEN,
      room:            'redacted',
    },
    'revision-data': {
      type: 'file-hash'
    },
  };

  if (deployTarget === 'development') {
    _merge(ENV, {
      build: {
        environment: 'development',
      },
      redis: {
        host: 'localhost',
      },
      s3: {
        prefix: 'development',
      }
    });
  }

  if (deployTarget === 'staging') {
    _merge(ENV, {
      build: {
        environment: 'staging',
      },
      redis: {
        host:   process.env.STAGING_REDIS_HOST || 'redacted',
      },
      s3: {
        prefix: 'staging',
      },
    });
  }

  if (deployTarget === 'production') {
    _merge(ENV, {
      build: {
        environment: 'production',
      },
      redis: {
        host: process.env.PRODUCTION_REDIS_HOST || 'redacted',
      },
      s3: {
        prefix: 'production',
      },
      'revision-data': {
        type: 'git-commit',
      },
    });
  }

  // Note: if you need to build some configuration asynchronously, you can return
  // a promise that resolves with the ENV object instead of returning the
  // ENV object synchronously.
  return ENV;
};
```

**Comment by achambers** [comment]
Ok, just identified the issue. So sorry about this guys. Will work on a fix shortly.

**Comment by achambers** [comment]
Released as [v0.4.2](https://github.com/ember-cli-deploy/ember-cli-deploy-redis/releases/tag/v0.4.2)
